### PR TITLE
Implement ReportConfig and related classes

### PR DESCRIPTION
**Pull request by kraglalbert**
# Summary

This is a big PR. Here are the main things it changes:

* Implement ReportConfig and ReportSectionConfig, which contain the config information for their respective non-config model classes
* Deleted ReportSection and replaced it with StudentReportSection and EmployerReportSection. This was done to reduce the amount of null data in report sections, since a report section would never be associated to both a Student and Employer report
* Made a `BaseController` class which contains an error handler that all controllers now extend 

Coming in future PRs:
* Service and integration tests for the new model classes
* #114 and #115 (unrelated to this PR, but I did some of that already here)

## Test Plan

`mvn test` and `mvn failsafe:integration-test` to make sure existing tests still pass

## Related Issues

#111

### CODE REVIEW

File: Backend/src/main/java/ca/mcgill/cooperator/controller/CourseController.java [hunk 1]
```diff
@@ -102,27 +95,4 @@ public CourseDto deleteCourse(@PathVariable int id) {
...
-    private List<CourseOffering> convertCourseOfferingListToDomainObject(
-            List<CourseOfferingDto> coDtos) {
-        List<CourseOffering> cos = new ArrayList<>(coDtos.size());
-        for (CourseOfferingDto coDto : coDtos) {
-            if (coDto != null) {
-                CourseOffering co = courseOfferingService.getCourseOfferingById(coDto.getId());
-                cos.add(co);
-            }
-        }
-        return cos;
```
Comment (emmaeagles) [inline Backend/src/main/java/ca/mcgill/cooperator/controller/CourseController.java]: why did you delete this?

File: Backend/src/main/java/ca/mcgill/cooperator/model/ReportConfig.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    public boolean getIsDeadlineFromStart() {
+        return this.isDeadlineFromStart;
+    }
+
+    public void setIsDeadlineFromStart(boolean isDeadlineFromStart) {
+        this.isDeadlineFromStart = isDeadlineFromStart;
+    }
+
```
Comment (emmaeagles) [inline Backend/src/main/java/ca/mcgill/cooperator/model/ReportConfig.java]: so if the admin modifies the due date its not longer from the start?

File: Backend/src/test/java/ca/mcgill/cooperator/controller/ControllerIT.java [hunk 3]
```diff
@@ -319,16 +329,17 @@ public EmployerContactDto createTestEmployerContact(
...
-    public ReportSectionDto createTestReportSection(StudentReportDto studentReportDto)
+    public StudentReportSectionDto createTestStudentReportSection(
+            ReportSectionConfigDto reportSectionConfigDto, StudentReportDto studentReportDto)
             throws Exception {
-        ReportSectionDto reportSectionDto = new ReportSectionDto();
+        StudentReportSectionDto reportSectionDto = new StudentReportSectionDto();
+        reportSectionDto.setResponse("I am a student and this is my response");
         reportSectionDto.setStudentReport(studentReportDto);
-        reportSectionDto.setTitle("Offer Letter Section");
-        reportSectionDto.setContent("spicing up my offer letter with a report section");
```
Comment (emmaeagles) [inline Backend/src/test/java/ca/mcgill/cooperator/controller/ControllerIT.java]: im really sad this is gone

File: Backend/src/main/java/ca/mcgill/cooperator/controller/CourseController.java (hunk 1 above)
Comment (kraglalbert) [reply Backend/src/main/java/ca/mcgill/cooperator/controller/CourseController.java]: I just moved this to `ControllerUtils` so that all of these types of methods are in one place

File: Backend/src/main/java/ca/mcgill/cooperator/model/ReportConfig.java (hunk 2 above)
Comment (kraglalbert) [reply Backend/src/main/java/ca/mcgill/cooperator/model/ReportConfig.java]: No, the fromStart refers to if the deadline is counting from the first day of the co-op (i.e. due 14 days from the first day of the co-op) or the last day (14 days before the last day). I could rename this if it's confusing

File: Backend/src/main/java/ca/mcgill/cooperator/controller/CourseController.java (hunk 1 above)
Comment (emmaeagles) [reply Backend/src/main/java/ca/mcgill/cooperator/controller/CourseController.java]: @kraglalbert you should make sure it still works because i had it there when i wrote this but the courseOfferingService was always null even though it was auto wired (i think bc the controller utils class is never defined with the auto wired thing in the classes that use it)

File: Backend/src/main/java/ca/mcgill/cooperator/model/ReportConfig.java (hunk 2 above)
Comment (emmaeagles) [reply Backend/src/main/java/ca/mcgill/cooperator/model/ReportConfig.java]: oh no its ok i just didnt know what was going on
